components/formats-common/src/loci/common/NIOFileHandle.java

```diff
@@ -263,12 +263,23 @@ public int read(ByteBuffer buf) throws IOException {
   @Override
   public int read(ByteBuffer buf, int off, int len) throws IOException {
```

Comment:
Noted due to the `int` cast below.  Why is the interface here using `int` rather than `long` for the offset and length?  Is is supposed to be restricted to small sizes?